### PR TITLE
feat: log all available addresses if listening host is 0.0.0.0

### PR DESCRIPTION
**Pull request by mcollina**
One of the few problems users have with listening to `0.0.0.0` manually is that they see `0.0.0.0` in their logs.

I'm converting it to _all_ interfaces `0.0.0.0` correspond to.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: fastify.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 const VERSION = '5.0.0-alpha.1'
 
+
```
Comment (jsumners) [inline fastify.js:5]:
```suggestion
```

File: lib/server.js [hunk 2]
```diff
@@ -297,23 +298,42 @@ function getServerInstance (options, httpHandler) {
   return server
 }
 
+function getAddresses (address) {
```
Comment (jsumners) [inline lib/server.js:301]:
```suggestion
/**
 * Inspects the provided `server.address` object and returns a
 * normalized list of IP address strings. Normalization in this
 * case refers to mapping wildcard `0.0.0.0` to the list of IP
 * addresses the wildcard refers to.
 *
 * @see https://nodejs.org/docs/latest/api/net.html#serveraddress
 *
 * @param {object} A server address object as described in the
 * linked docs.
 *
 * @returns {string[]}
 */
function getAddresses (address) {
```

**Review by jsumners — APPROVED** [review]
This will be a great DX improvement.

**Review by metcoder95 — APPROVED** [review]

**Comment by mcollina** [comment]
Is this backportable to v4?

**Comment by jsumners** [comment]
Seems plausible and reasonable to me.

**Comment by gurgunday** [comment]
Coverage is failing

**Comment by mcollina** [comment]
PTAL

**Comment by mcollina** [comment]
Does anybody has a clue on:

Screenshot 2024-05-21 at 19 01 49

**Comment by mcollina** [comment]
Because GitHub had an outage.

**Pull request merged**